packer/deploy-ami.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+# Copyright 2017 by the contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# This script is for after you've run create-ami.sh and want to make the AMI
+# public, and copy it to all AWS regions.
+
+set -o errexit
+set -o pipefail
+
+log_cat() {
+  if [[ "${VERBOSE}" == "true" ]]; then
+    cat 1>&2
+  fi
+}
+
+log() {
+  if [[ "${VERBOSE}" == "true" ]]; then
+    echo $* 1>&2
+  fi
+}
+
+usage() {
+  log "Usage: $0 -r SOURCE_REGION -i SOURCE_AMI [-q]"
+  log_cat 1>&2 <<EOF
+
+Will copy SOURCE_AMI from SOURCE_REGION to all other regions, and will mark the
+image as public in all regions.  This is intended to be run with the AMI that
+packer produces, with the region specified in create-ami.sh
+
+Options:
+	-q: don't output status messages, just the resulting YAML.
+EOF
+}
+
+VERBOSE="true"
+while getopts "r:i:q:" opt; do
+  case $opt in
+    r)
+      SOURCE_REGION=$OPTARG
+      ;;
+    i)
+      SOURCE_AMI=$OPTARG
+      ;;
+    q)
+      VERBOSE=''
+      ;;
+    \?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "${SOURCE_REGION}" || -z "${SOURCE_AMI}" ]]; then
+  usage
+  exit 1
+fi
+
+DEST_REGIONS=$(aws ec2 describe-regions --output json | jq -r '.Regions[].RegionName' | grep -v "^${SOURCE_REGION}$")
```

Comment:
I think I'm missing something, because isn't that what this command is doing?  I'm grabbing all regions everywhere, and `grep -v`'ing the original region, resulting in a list of all regions but the one it came from.